File: src/org/emulinker/util/GameDataCache.java
package org.emulinker.util;

public interface GameDataCache
{
	public byte[] get(int index);

	public int add(byte[] data);

	public int indexOf(byte[] data);

	public int size();

	public boolean isEmpty();

	public void clear();

	public boolean contains(byte[] data);

	public byte[] set(int index, byte[] data);

	public byte[] remove(int index);
}

File: emulinker/src/main/java/org/emulinker/util/GameDataCache.kt
package org.emulinker.util

interface GameDataCache {
  val isEmpty: Boolean
  val size: Int

  operator fun get(index: Int): ByteArray?
  operator fun contains(data: ByteArray?): Boolean
  operator fun set(index: Int, data: ByteArray?): ByteArray?

  fun add(data: ByteArray?): Int
  fun indexOf(data: ByteArray?): Int
  fun clear()
  fun remove(index: Int): ByteArray?
}
